arelle/plugin/validate/DBA/rules/tm.py

```diff
@@ -278,7 +278,11 @@ def rule_tm31(
     """
     DBA.TM31: gsd:DateOfApprovalOfReport must only be tagged once if tagged
     """
-    return errorOnMultipleFacts(val.modelXbrl, pluginData.dateOfApprovalOfReportQn, 'DBA.TM31')
+    reportTypeFacts = val.modelXbrl.factsByQname.get(pluginData.informationOnTypeOfSubmittedReportQn, set())
```

Comment:
Since this same logic is going to preclude at least a few validations from running at all, might be nice to write a helper function so we can just do:
```
if not pluginData.isAnnualReport(val.modelXbrl):
	return
```

Reply:
I have one last PR to make and Ill simplify all the code in that one.